visual-python/src/saucelabs_visual/client.py

```diff
@@ -0,0 +1,185 @@
+
+    def create_build(
+            self,
+            name: str = environ.get('SAUCE_VISUAL_BUILD_NAME') or None,
+            project: str = environ.get('SAUCE_VISUAL_PROJECT_NAME') or None,
+            branch: str = environ.get('SAUCE_VISUAL_BRANCH_NAME') or None,
+            default_branch: str = environ.get('SAUCE_VISUAL_DEFAULT_BRANCH_NAME') or None,
+            custom_id: str = environ.get('SAUCE_VISUAL_CUSTOM_ID') or None,
+            keep_alive_timeout: int = None
+    ):
+        query = gql(
+            # language=GraphQL
+            """
+            mutation createBuild(
+                $name: String!,
+                $project: String,
+                $branch: String,
+                $customId: String,
+                $defaultBranch: String,
+                $keepAliveTimeout: Int,
+            ) {
+                createBuild(input: {
+                    name: $name,
+                    project: $project,
+                    branch: $branch,
+                    customId: $customId,
+                    defaultBranch: $defaultBranch,
+                    keepAliveTimeout: $keepAliveTimeout,
+                }){
+                    id
+                    url
+                }
+            }
+            """
+        )
+        values = {
+            "name": name,
+            "project": project,
+            "branch": branch,
+            "customId": custom_id,
+            "defaultBranch": default_branch,
+            "keepAliveTimeout": keep_alive_timeout,
+        }
+        build = self.client.execute(query, variable_values=values)
+        self.build_id = build['createBuild']['id']
+        return build
+
+    def finish_build(self):
+        query = gql(
+            # language=GraphQL
+            """
+            mutation finishBuild($id: UUID!) {
+                finishBuild(input: {uuid: $id}){
+                    id
+                    url
+                    status
+                }
+            }
+            """
+        )
+        values = {"id": self.build_id}
+        return self.client.execute(query, variable_values=values)
+
+    def get_selenium_metadata(self, session_id: str):
+        query = gql(
+            # language=GraphQL
+            """
+            query webdriverSessionInfo(
+                $jobId: ID!,
+                $sessionId: ID!
+            ) {
+                meta: webdriverSessionInfo(input: {
+                    sessionId: $sessionId,
+                    jobId: $jobId,
+                }){
+                    blob
+                    operatingSystem
+                    operatingSystemVersion
+                    browser
+                    browserVersion
+                    deviceName
+                }
+            }
+            """
+        )
+        values = {"sessionId": session_id, "jobId": session_id}
+        meta = self.client.execute(query, variable_values=values)
+        return meta['meta']['blob']
+
+    def create_snapshot_from_webdriver(
```

Comment:
1. how about more generic function name? In the end it doesn't matter what is behind the method.
2. Should we consider passing some options via option object?

Reply:
This one specifically requires a running selenium session through Sauce for us to execute the backend snapshot tasks. More generic `create_snapshot` and `upload_snapshot` methods will likely come later once we broaden the framework support.